packages/victory-core/src/victory-util/data.js

```diff
@@ -91,6 +92,72 @@ function cleanData(dataset, props) {
   });
 }
 
+// This method will remove data points that fall outside of the desired domain (non-continuous charts only)
+function formatDataFromDomain(dataset, props) {
+  const { domain, symbol, interpolation } = props;
```

Comment:
I'm not sure, but I think you might want to use `Domain.getMinFromProps(props, axis)` and `Domain.getMaxFromProps(props, axis)` (where axis is "x" or "y") so that you are sure to be capturing information from the `domain`, `minDomain` and `maxDomain` props.

Reply:
I could be wrong, but based on my testing, domain here already has the prop values calculated into it, so getting them again from the props seems redundant.

Reply:
I ending up using `getMin/MaxFromProps` to solve the nulling of points without a VictoryChart as you described here:


> When you have the domain defined directly on a component like VictoryScatter not wrapped in VictoryChart you can end up in a situation where you don't have a domain prop that has already reconciled other props like minDomain, because getData is called before getDomain (as the data may inform the domain). Check out a couple examples:
> 
> This one wont null out the point at (1, 2)
> ```
> <VictoryScatter
>   minDomain={{ y: 2.2 }}
>   data={[
>     { x: 0, y: 3 },
>     { x: 1, y: 2 },
>     { x: 2, y: 4 },
>   ]}
> />
> ```
> 
> And actually neither will this:
> 
> ```
> <VictoryScatter
>   domain={{ y: [2.2, 4] }}
>   data={[
>     { x: 0, y: 3.5 },
>     { x: 1, y: 2 },
>     { x: 2, y: 4 },
>   ]}
> />
> ```